
# Curated list of papers on variation with SARS-CoV2

(last update Sept 10, 2020)

https://www.ncbi.nlm.nih.gov/pubmed/32752979

Andrés C, Garcia-Cehic D, Gregori J, Piñana M, Rodriguez-Frias F, Guerrero-Murillo M, Esperalba J, Rando A, Goterris L, Codina MG, Quer S, Martín MC, Campins M, Ferrer R, Almirante B, Esteban JI, Pumarola T, Antón A, Quer J. Naturally occurring SARS-CoV-2 gene deletions close to the spike S1/S2 cleavage site in the viral quasispecies of COVID19 patients. Emerg Microbes Infect. 2020 Dec;9(1):1900–1911. PMID: 32752979

This study identified large number of deletion in S gene

Sequencing of amplified S gene fragments using Illumina (miSeq) PRJNA630679

Deletions, Amplicon, Illumina, S gene


-----

http://dx.doi.org/10.1080/22221751.2020.1782271

Gong Y-N, Tsao K-C, Hsiao M-J, Huang C-G, Huang P-N, Huang P-W, Lee K-M, Liu Y-C, Yang S-L, Kuo R-L, Chen K-F, Liu Y-C, Huang S-Y, Huang H-I, Liu M-T, Yang J-R, Chiu C-H, Yang C-T, Chen G-W, Shih S-R. SARS-CoV-2 genomic surveillance in Taiwan revealed novel ORF8-deletion mutant and clade possibly associated with infections in Middle East. Emerg Microbes Infect. 2020 Dec;9(1):1457–1466. PMID: 32543353

This study identified deletion of ORF8 in one Taiwanese sample. Other mutations identified are ORF1ab-V378I (novel), ORF8-L84S, ORF3a-G251V, and S-D614G.

WGS using Illumina, GISAID, no Read data

Deletion, ORF1ab-V378I, ORF8-L84S, ORF3a-G251V, and S-D614G, WGS

------

http://dx.doi.org/10.1038/d41586-020-02544-6

Callaway E. The coronavirus is mutating — does it matter? Nature. Nature Publishing Group; 2020 Sep 8;585(7824):174–177.

An opinion piece of the merits of the D614G and its implications

Opinion, Review

-------

http://dx.doi.org/10.1016/j.cell.2020.07.012

Li Q, Wu J, Nie J, Zhang L, Hao H, Liu S, Zhao C, Zhang Q, Liu H, Nie L, Qin H, Wang M, Lu Q, Li X, Sun Q, Liu J, Zhang L, Li X, Huang W, Wang Y. The Impact of Mutations in SARS-CoV-2 Spike on Viral Infectivity and Antigenicity. Cell. 2020 Sep 3;182(5):1284–1294.e9. PMCID: PMC7366990

![](https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/attachment/689a6d1a-07f1-44dc-9fb3-3644ad204a42/gr3.jpg)

Infectivity analysis of 106 S gene mutants (substitutions and glycosilation alterations) identified from GISAID sequences (May 6, 2020). The dominant D614G itself and combined with other
mutations are more infectious. Ablation of both N331 and N343 glycosylation at RBD
drastically reduced infectivity. Ten mutations such as N234Q, L452R, A475V, and V483A
was markedly resistant to some mAbs

Mutations effect analysis, S gene

--------

http://dx.doi.org/10.1016/j.cell.2020.08.012

Starr TN, Greaney AJ, Hilton SK, Ellis D, Crawford KHD, Dingens AS, Navarro MJ, Bowen JE, Tortorici MA, Walls AC, King NP, Veesler D, Bloom JD. Deep Mutational Scanning of SARS-CoV-2 Receptor Binding Domain Reveals Constraints on Folding and ACE2 Binding. Cell. 2020 Sep 3;182(5):1295–1310.e20. PMCID: PMC7418704

![](https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/attachment/01b3c8e1-9343-4e0a-9f40-887cdd50f33e/gr3.jpg)

Deep mutation scanning of all possible S gene (receptor binding domain) mutations and their affect on (1) expression and (2) affinity towards ACE2. This gives detailed function info about S protein amino acids 331 - 531.

Mutation effect analysis, S gene, RBD

-------

http://dx.doi.org/10.1101/gr.266221.120

Gomez-Carballa A, Bello X, Pardo-Seco J, Martinon-Torres F, Salas A. Mapping genome variation of SARS-CoV-2 worldwide highlights the impact of COVID-19 super-spreaders. Genome Res [Internet]. 2020 Sep 2; Available from: http://dx.doi.org/10.1101/gr.266221.120 PMID: 32878977

Analysis of GISAID data to identify temporal progression of SARS-VCov2 clades. Identifies 12 Nov 2019 as the TMCA for SARS-CoV2.

Clades, TMCA, Genomic diversity

-------

http://dx.doi.org/10.1016/j.meegid.2020.104525

Pereira F. Evolutionary dynamics of the SARS-CoV-2 ORF8 accessory gene. Infect Genet Evol. 2020 Sep 2;85:104525. PMCID: PMC7467077

"*These findings suggest that SARS-CoV-2 can persist without a functional ORF8 protein. Deletion breakpoints were found located in predicted hairpins suggesti g a possible involvement of these elements in the rearrangement process. Although the function of ORF8 remains to be elucidated, its structural plasticity and high diversity suggest an important role in SARS-CoV-2 pathogenicity.*"


Orf8

-----

http://dx.doi.org/10.1016/j.meegid.2020.104351

van Dorp L, Acman M, Richard D, Shaw LP, Ford CE, Ormond L, Owen CJ, Pang J, Tan CCS, Boshier FAT, Ortiz AT, Balloux F. Emergence of genomic diversity and recurrent mutations in SARS-CoV-2. Infect Genet Evol. 2020 Sep;83:104351. PMCID: PMC7199730

Analyzed GISAID data (April 19, 2020) and user SRA data to confirm some of the variable sites. This is a comprehensive analysis of variation progression from GISAID data. Cites SKP observable notebook.

Clades, TMCA, Genomic diversity

------

http://dx.doi.org/10.1016/j.ygeno.2020.04.016

Yin C. Genotyping coronavirus SARS-CoV-2: methods and implications. Genomics. 2020 Sep;112(5):3588–3596. PMCID: PMC7184998

A somewhat simplistic way to catalog SNP variation across early set of GISAID data.

Clades, TMCA, Genomic diversity

------

http://dx.doi.org/10.1073/pnas.2008281117

Dearlove B, Lewitus E, Bai H, Li Y, Reeves DB, Joyce MG, Scott PT, Amare MF, Vasan S, Michael NL, Modjarrad K, Rolland M. A SARS-CoV-2 vaccine candidate would likely match all currently circulating variants. Proc Natl Acad Sci U S A [Internet]. 2020 Aug 31; Available from: http://dx.doi.org/10.1073/pnas.2008281117 PMID: 32868447

Analysis of GISAID genomes (May 18, 2020) to understand whether global variation will affect vaccine efficacy. They did not find evidence that variation observed so far will create issues for vaccine design.

Genomic Diversity

-------

http://dx.doi.org/10.1101/2020.08.28.271965

Schlick T, Zhu Q, Jain S, Yan S. Structure-Altering Mutations of the SARS-CoV-2 Frame Shifting RNA Element. bioRxiv [Internet]. 2020 Aug 30; Available from: http://dx.doi.org/10.1101/2020.08.28.271965 PMCID: PMC7457599

The authors identify a minimum set of mutations that destroy programmed frame shifting required for orf1ab transcription: "*only two residue changes can accomplish a dramatic change in structure, namely destruction of Stem 2 and the pseudoknot*"

Fremeshifting

---------

http://dx.doi.org/10.1371/journal.pone.0238490

Li Y, Yang X, Wang N, Wang H, Yin B, Yang X, Jiang W. SNPs or RNA modifications? Concerns on mutation-based evolutionary studies of SARS-CoV-2. PLoS One. 2020 Aug 28;15(8):e0238490. PMCID: PMC7454988

Attempt (and a cautionary note) to highlight the fact that RNA modification can be responsible for some SNPs observed in SARS-CoV2 data.

RNA editing

-------

http://dx.doi.org/10.1016/j.virol.2020.08.010

Jones LR, Manrique JM. Quantitative phylogenomic evidence reveals a spatially structured SARS-CoV-2 diversity. Virology. 2020 Aug 26;550:70–77. PMCID: PMC7448746

Analysis of SARS-CoV2 global dispersion patterns based on GISAID (April 25, 2020) data.

Clades, Genomic diversity

-------

http://dx.doi.org/10.1101/2020.08.26.267831

Siqueira JD, Goes LR, Alves BM, de Carvalho PS, Cicala C, Arthos J, Viola JPB, de Melo AC, Soares MA. SARS-CoV-2 genomic and quasispecies analyses in cancer patients reveal relaxed intrahost virus evolution. bioRxiv [Internet]. 2020 Aug 26; Available from: http://dx.doi.org/10.1101/2020.08.26.267831 PMCID: PMC7457605

Analysis of intra-host SARS-CoV2 variation in cancer patients and non-cancer controls. Cancer patents have higher intra-host variability. 

Sequencing of ARTIC amplicons. Illumina sequencing. No data is provided.

Intrahost, Illumina, Amplicon, ARTIC, Nodata

------

http://dx.doi.org/10.1016/j.ijid.2020.08.052

Guan Q, Sadykov M, Mfarrej S, Hala S, Naeem R, Nugmanova R, Al-Omari A, Salih S, Mutair AA, Carr MJ, Hall WW, Arold ST, Pain A. A genetic barcode of SARS-CoV-2 for monitoring global distribution of different clades during the COVID-19 pandemic. Int J Infect Dis [Internet]. 2020 Aug 22; Available from: http://dx.doi.org/10.1016/j.ijid.2020.08.052 PMCID: PMC7443060

Development of a genotyping protocol that takes advantage of known SARS-CoV2 variants to quickly identify clades. 


Clades, Genomic diversity





